sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -199,6 +209,19 @@ private[sql] case class FileSourceScanExec(
         options = relation.options,
         hadoopConf = relation.sparkSession.sessionState.newHadoopConfWithOptions(relation.options))
 
+      (file: PartitionedFile) => {
+        val iter = func(file)
+        // Only for test purpose.
+        // Once the vectorized Parquet reader is initialized in the above method, we can read its
+        // variable numRowGroups.
+        if (fileFormat != null) {
```

Comment:
I think we can directly update the accumulator in ParquetFileFormat if there is the accumulator. I will update the codes later and please check if it is good for you.